### PR TITLE
Realtime RC Fetch ID

### PR DESCRIPTION
**Pull request by qdpham13**
Implementation of [go/realtime-rc-fetch-id](http://goto.google.com/realtime-rc-fetch-id)

### CODE REVIEW

File: FirebaseRemoteConfig/Tests/Unit/RCNPersonalizationTest.m [hunk 1]
```diff
@@ -243,10 +245,13 @@ + (id)mockFetchRequest {
...
         __unsafe_unretained FIRRemoteConfigFetchCompletion handler;
         [invocation getArgument:&handler atIndex:3];
-        [configFetch fetchWithUserProperties:[[NSDictionary alloc] init] completionHandler:handler];
+        [configFetch fetchWithUserProperties:[[NSDictionary alloc] init]
+                             fetchTypeHeader:@"BaseFetch/1"
+                           completionHandler:handler];
       });
   OCMExpect([configFetch
                 URLSessionDataTaskWithContent:[OCMArg any]
+                              fetchTypeHeader:@"BaseFetch/1"
```
Comment (karenyz) [inline FirebaseRemoteConfig/Tests/Unit/RCNPersonalizationTest.m:254]: Is it "BaseFetch" or "Base"?

File: FirebaseRemoteConfig/Sources/RCNConfigFetch.m [hunk 2]
```diff
@@ -191,7 +196,84 @@ - (void)fetchConfigWithExpirationDuration:(NSTimeInterval)expirationDuration
...
     }
     strongSelf->_settings.isFetchInProgress = YES;
-    [strongSelf refreshInstallationsTokenWithCompletionHandler:completionHandler];
+    NSString *fetchTypeHeader = [NSString stringWithFormat:@"%@/1", kBaseFetchType];
+    [strongSelf refreshInstallationsTokenWithCompletionHandler:fetchTypeHeader
+                                             completionHandler:completionHandler];
+  });
+}
+
+- (void)realtimeFetchConfigWithExpirationDuration:(NSTimeInterval)expirationDuration
```
Comment (karenyz) [inline FirebaseRemoteConfig/Sources/RCNConfigFetch.m:205]: does this method need `expirationDuration`? for realtime i think it will always be set to 0?

**Review by karenyz — CHANGES_REQUESTED** [review]

File: FirebaseRemoteConfig/Tests/Unit/RCNPersonalizationTest.m (hunk 1 above)
Comment (qdpham13) [reply FirebaseRemoteConfig/Tests/Unit/RCNPersonalizationTest.m]: Yeah it's base, sorry. Updates.

File: FirebaseRemoteConfig/Sources/RCNConfigFetch.m (hunk 2 above)
Comment (qdpham13) [reply FirebaseRemoteConfig/Sources/RCNConfigFetch.m]: That's true, it doesn't need it. Removed and updated naming.

File: FirebaseRemoteConfig/Sources/RCNConfigRealtime.m [hunk 3]
```diff
@@ -432,37 +432,40 @@ - (void)fetchLatestConfig:(NSInteger)remainingAttempts targetVersion:(NSInteger)
...
-                              } else {
-                                FIRLogDebug(
-                                    kFIRLoggerRemoteConfig, @"I-RCN000016",
-                                    @"Fetched config's template version is outdated, re-fetching");
-                                [strongSelf autoFetch:remainingAttempts - 1
-                                        targetVersion:targetVersion];
-                              }
-                            }
-                          }];
+          realtimeFetchConfigWithNoExpirationDuration:gFetchAttempts - remainingAttempts + 1
```
Comment (karenyz) [inline FirebaseRemoteConfig/Sources/RCNConfigRealtime.m:435]: nit: you could just decrement `remainingAttempts` at the beginning of the method, then you don't need the "+1" here and you can directly pass `remainingAttempts` to the next autofetch call

**Review by karenyz — APPROVED** [review]

**Pull request merged**